internals/manifestworkSync/controller/cappPlacementSync_controller.go

```diff
@@ -83,29 +76,19 @@ var CappPredicateFuncs = predicate.Funcs{
 // SyncManifestWork checks whether the manifest work deploying the Capp exists in the managed cluster namespace
 // If it does, it updates the Capp in the manifest work spec. If it doesn't then it creates it
 func (r *CappNamespaceReconciler) SyncManifestWork(capp rcsv1alpha1.Capp, ctx context.Context, logger logr.Logger) (ctrl.Result, error) {
-	mwName := utils.NamespaceManifestWorkPrefix + capp.Namespace + "-" + capp.Name
+	mwName := adapters.GenerateMwName(capp)
 	managedClusterName := capp.Annotations[utils.AnnotationKeyHasPlacement]
 	var mw workv1.ManifestWork
-	manifests, err := utils.GatherCappResources(capp, ctx, logger, r.Client)
+	cappDirector := director.CappDirector{Ctx: ctx, K8sclient: r.Client, Log: logger, EventRecorder: r.EventRecorder}
+	manifests, err := cappDirector.AssembleManifests(capp)
 	if err != nil {
-		r.EventRecorder.Event(&capp, eventTypeError, eventCappVolumeNotFound, err.Error())
-		if err := statusutils.SetVolumesCondition(capp, ctx, r.Client, logger, false, err.Error()); err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to set volume condition: %s", err.Error())
-		}
-		return ctrl.Result{}, fmt.Errorf("failed to get one of the volumes from Capp spec: %s", err.Error())
+		return ctrl.Result{}, fmt.Errorf("failed to build ManifestWork: %s", err.Error())
```

Comment:
You need to use %v when returning error strings (https://google.github.io/styleguide/go/decisions#error-strings)